datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -990,24 +989,44 @@ pub async fn from_substrait_rel(
 fn ensure_schema_compatability(
     table: DataFrame,
     substrait_schema: DFSchema,
-) -> Result<DataFrame> {
-    let df_schema = table.schema().to_owned().strip_qualifiers();
+) -> Result<LogicalPlan> {
+    let df_schema = table.schema().to_owned();
     if df_schema.logically_equivalent_names_and_types(&substrait_schema) {
-        return Ok(table);
+        return Ok(table.into_unoptimized_plan());
     }
-    let selected_columns = substrait_schema
-        .strip_qualifiers()
-        .fields()
-        .iter()
-        .map(|substrait_field| {
-            let df_field =
-                df_schema.field_with_unqualified_name(substrait_field.name())?;
-            ensure_field_compatability(df_field, substrait_field)?;
-            Ok(col(format!("\"{}\"", df_field.name())))
-        })
-        .collect::<Result<_>>()?;
 
-    table.select(selected_columns)
+    let t = table.into_unoptimized_plan();
```

Comment:
nit: move this up and then do Ok(t) inside the `if`?

Reply:
done